### PR TITLE
chore(stock): improve stock settings terminology

### LINKED ISSUE #6520: Make sure we refer to stock exits with the same terminology everywhere
In the depot management form, we refer to the stock exits with the terminology "exit to individuals" but we often use the term "exit to patients".  We need to ensure that we use the same language everywhere so that we don't confuse our users.  In particular, our stock exit reports, the stock exit module, etc.

image

### PR DESCRIPTION
**Pull request by jniles**
This commit rewrites and improves both the layout and language used on stock settings, the depot permission modal, and a few select locations to ensure that the English text is clear.

Closes #6520.

Here is what it looks like now:
image

### CODE REVIEW

File: client/src/i18n/en/stock.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "STOCK" : {
     "ALERT" : "Alert",
-    "USE_OF_STOCK_WAC" : "Use of the value in stock of the product (Weighted Average Cost)",
+    "USE_OF_STOCK_WAC" : "This value is calulated automatically using the Weighted Average Cost (WAC) algorithm.",
```
Comment (jmcameron) [inline client/src/i18n/en/stock.json:4]: 👍

File: client/src/i18n/en/stock.json [hunk 2]
```diff
@@ -35,7 +35,7 @@
       "OVER_MAX"      : "Products over maximum",
       "PRODUCTS_AT_RISK_EXPIRATION": "Products at risk of expiration",
       "PRODUCTS_AT_RISK_OUT_STOCK" : "Products at risk of out of stock",
-      "PRODUCTS_IN_OVERSTOCK" : "Products in overstock"
+      "PRODUCTS_IN_OVERSTOCK" : "Products overstocked"
```
Comment (jmcameron) [inline client/src/i18n/en/stock.json:38]: 👍

File: client/src/i18n/en/stock.json [hunk 3]
```diff
@@ -407,10 +405,10 @@
     "EDIT_ASSIGN" : "Edit Assignment",
     "ADD_ASSIGNMENT" : "Add Assignment",
     "REMOVE_ASSIGNMENT" : "Remove Assignment",
-    "CREATE_SUCCESS" : "Successfully created",
-    "UPDATE_SUCCESS" : "Successfully updated",
-    "REMOVE_SUCCESS" : "Successfully removed",
-    "CONFIRM_REMOVE_MSG" : "Do you really want to remove this stock assignment ?",
+    "CREATE_SUCCESS" : "Successfully created the assignment",
```
Comment (jmcameron) [inline client/src/i18n/en/stock.json:408]: Nice.  Friendlier language!

File: server/controllers/stock/core.js [hunk 4]
```diff
@@ -698,8 +698,9 @@ function listLostStock(params) {
  *   computeLotIndicators()
  *
  * DEFINITIONS:
- *   S_SEC: Security Stock - one month of stock on hand based on the average consumption.
- *   S_MIN: Minimum stock - typically the security stock (depends on the depot)
+ *   S_SEC: Security Stock - the amount of stock needed to survive a reorder period without stockout.
+ *                           this depends on the purchase/delivery delay.
+ *   S_MIN: Minimum stock - typically the security stock (depends on the depot), plus some multiplier.
```
Comment (jmcameron) [inline server/controllers/stock/core.js]: Instead of 'plus', perhaps say "with"?

File: server/controllers/stock/core.js [hunk 5]
```diff
@@ -730,14 +731,12 @@ function computeInventoryIndicators(inventories) {
     inventory.S_SEC = CMM * delay; // stock de securite
 
     // Compute Minimum Stock
-    // The minumum of stock required is double the security stock.
-    // NOTE(@jniles): this is defined per depot.
+    // The minimum stock is used to trigger the "minimum stock" warning state.  Note that this
+    // value can be less than the security stock if the min_months_security_stock is less than one.
+    // FIXME(jniles): rename min_months_security_stock variable since it doesn't depend on months
```
Comment (jmcameron) [inline server/controllers/stock/core.js:736]: Can we fix this in this PR?

File: client/src/i18n/en/stock.json [hunk 6]
```diff
@@ -299,18 +297,18 @@
     "WAC" : "Weighted Avgerage Cost"
   },
   "SETTINGS" : {
-    "DEFAULT_MIN_MONTHS_SECURITY_STOCK": "Default minimum number of months of security stock for depots",
-    "DEFAULT_MIN_MONTHS_SECURITY_STOCK_HELP_TEXT": "The stock supply should provide at least this many months of stock for the depot involved.",
-    "DEFAULT_PURCHASE_ORDER_INTERVAL" : "Default Purchase order interval (In Month)",
-    "DEFAULT_PURCHASE_ORDER_INTERVAL_TEXT" : "Sets the default minimum order interval",
-    "DEFINE_MONTH_AVERAGE_CONSUMPTION": "Number of months for calculating the average monthly consumption",
+    "DEFAULT_MIN_MONTHS_SECURITY_STOCK": "Security stock multiplier",
+    "DEFAULT_MIN_MONTHS_SECURITY_STOCK_HELP_TEXT": "This value is multiplied by the security stock to determine the minimum quantity of stock to keep in the depot (Smin) using the formula (Smin = SS * ss_multiplier).  Note that values greater than one will cause the minimum stock alert to be greater than the security stock, and less than one will cause it to be less.",
```
Comment (jmcameron) [inline client/src/i18n/en/stock.json:301]: Suggestion:
- Change "multiplied by the security stock" to "multiplied by the value of the security stock (SS)"...

File: server/controllers/stock/core.js [hunk 7]
```diff
@@ -698,8 +698,9 @@ function listLostStock(params) {
  *   computeLotIndicators()
  *
  * DEFINITIONS:
- *   S_SEC: Security Stock - one month of stock on hand based on the average consumption.
- *   S_MIN: Minimum stock - typically the security stock (depends on the depot)
+ *   S_SEC: Security Stock - the amount of stock needed to survive a reorder period without stockout.
```
Comment (jmcameron) [inline server/controllers/stock/core.js:701]: 👍

**Review by jmcameron — CHANGES_REQUESTED** [review]
I reviewed the language changes and most were excellent.  I added some suggestions for a few of them.   

I found a translation problem in the Stock Requisition create/edit modals:
image

I tested this and reviewed the stock settings page.   It seemed fine.  However, I suggest:
- Move "Minimum delivery delay" setting above "Security stock multiplier" since the MDD is used to compute the value that SSM is applied to.


There are a couple of other missing translation items in the translations. These are not related to this PR, but this is good chance to fix them anyway.  Or we can add an issue to fix them.

```
   $ node utilities/translation/tfcomp.js client/src/i18n/en client/src/i18n/fr
   Missing from /home/jmcameron/Bhima/client/src/i18n/fr/asset.json:
     ASSET.SHOW_ONLY_LAST_SCAN

   Missing from /home/jmcameron/Bhima/client/src/i18n/en/report.json:
     REPORT.STOCK_VALUE.THROUGH

   Missing from /home/jmcameron/Bhima/client/src/i18n/fr/report.json:
     REPORT.THROUGH
```

File: server/controllers/stock/core.js (hunk 5 above)
Comment (jniles) [reply server/controllers/stock/core.js:736]: I prefer to do the translation changes and DB changes in two different PRs.  I'll also improve the requisition performance.

**Comment by jniles** [comment]
@jmcameron this is ready to be reviewed again.

File: client/src/i18n/en/stock.json (hunk 1 above)
Comment (jmcameron) [reply client/src/i18n/en/stock.json:4]: Looks good

File: server/controllers/stock/core.js (hunk 5 above)
Comment (jmcameron) [reply server/controllers/stock/core.js:736]: Sounds good

**Review by jmcameron — APPROVED** [review]
Checked the changes and they all look good.

**Pull request merged**